src/controllers/ConfirmationController.ts

```diff
@@ -18,10 +18,10 @@ export class ConfirmationController extends BaseAsyncHttpController {
 
         const companyNumber = req.session
             .chain(_ => _.getExtraData())
-            .chainNullable(data => data[AppealKeys.APPEALS_KEY])
-            .chainNullable(appeals => appeals[AppealKeys.PENALTY_IDENTIFIER])
-            .map(penaltyIdentifier => penaltyIdentifier[PenaltyIdentifierKeys.COMPANY_NUMBER])
-            .extract();
+            .chainNullable(data => data.appeals as Appeal)
+            .chainNullable(appeals => appeals.penaltyIdentifier as PenaltyIdentifier)
+            .map(penaltyIdentifier => penaltyIdentifier.companyNumber)
+            .orDefault('');
```

Comment:
Use of `extract` made more sense. At this stage value is expected to be present in the session. If not an error should be thrown. There should be a check added in the future to fail fast with an error page. For now revert please.